### PR TITLE
Fix for when_snapshotStartedBeforeExecution test

### LINKED ISSUE #641: com.hazelcast.jet.core.JobRestartWithSnapshotTest.when_snapshotStartedBeforeExecution_then_firstSnapshotIsSuccessful
```Error Message

snapshot was not successful
Stacktrace

java.lang.AssertionError: snapshot was not successful
	at com.hazelcast.jet.core.JobRestartWithSnapshotTest.lambda$when_snapshotStartedBeforeExecution_then_firstSnapshotIsSuccessful$5(JobRestartWithSnapshotTest.java:335)
	at com.hazelcast.jet.core.JobRestartWithSnapshotTest.when_snapshotStartedBeforeExecution_then_firstSnapshotIsSuccessful(JobRestartWithSnapshotTest.java:331)
```

https://hazelcast-l337.ci.cloudbees.com/view/Jet/job/Jet/com.hazelcast.jet$hazelcast-jet-core/581/testReport/junit/com.hazelcast.jet.core/JobRestartWithSnapshotTest/when_snapshotStartedBeforeExecution_then_firstSnapshotIsSuccessful/

### PR DESCRIPTION
**Pull request by cangencer**
* Increase timeout
* Prevent further snapshots from being completed to prevent
  snapshot record being deleted.

Fixes #641

### CODE REVIEW

File: hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/JobRestartWithSnapshotTest.java [hunk 1]
```diff
@@ -315,12 +315,7 @@ public void when_snapshotStartedBeforeExecution_then_firstSnapshotIsSuccessful()
...
-        SequencesInPartitionsMetaSupplier pms = new SequencesInPartitionsMetaSupplier(3, 100);
-
-        Vertex source = dag.newVertex("source", throttle(pms, 10)).localParallelism(1);
-        Vertex sink = dag.newVertex("sink", writeListP("sink"));
-
-        dag.edge(between(source, sink).distributed());
+        dag.newVertex("p", FirstSnapshotProcessor::new).localParallelism(1);
 
         JobConfig config = new JobConfig();
         config.setProcessingGuarantee(ProcessingGuarantee.EXACTLY_ONCE);
```
Comment (viliam-durina) [inline hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/JobRestartWithSnapshotTest.java]: We can now set the snapshot interval to 0 to make the test even more reliable.

**Comment by cangencer** [comment]
verify

**Comment by cangencer** [comment]
verify

**Comment by cangencer** [comment]
verify

**Comment by cangencer** [comment]
verify

**Pull request merged**